test/test_sde.py

```diff
@@ -0,0 +1,245 @@
+# at dt=2**-4 than Euler is at dt=2**-14 (and it takes forever to run at such
+# a small dt). Hence , the order of convergence of ShARK seems to plateau at
+# discretisations finer than 2**-4.
+# I propose the following:
+# We can sparate this test into two. First we determine how fast the solver
+# converges to its own limit (i.e. using itself as reference), and then
+# check whether that limit is the same as the Euler/Heun limit.
+# For the second, I would like to make a separate check, where the "correct"
+# solution is computed only once and then all solvers are compared to it.
+@pytest.mark.parametrize("solver_ctr,noise,theoretical_order", _solvers_and_orders())
+def test_sde_strong_order_new(
+    solver_ctr, noise: Literal["any", "com", "add"], theoretical_order
+):
+    bmkey = jr.PRNGKey(5678)
+    sde_key = jr.PRNGKey(11)
+    num_samples = 100
+    bmkeys = jr.split(bmkey, num=num_samples)
+    t0 = 0.3
+    t1 = 5.3
+
+    if noise == "add":
+        sde = get_time_sde(t0, t1, jnp.float64, sde_key, noise_dim=7)
+    else:
+        if noise == "com":
+            noise_dim = 1
+        elif noise == "any":
+            noise_dim = 5
+        else:
+            assert False
+        sde = get_mlp_sde(t0, t1, jnp.float64, sde_key, noise_dim=noise_dim)
+
+    ref_solver = solver_ctr()
+    level_coarse, level_fine = 4, 10
+
+    # We specify the times to which we step in way that each level contains all the
+    # steps of the previous level. This is so that we can compare the solutions at
+    # all the times in saveat, and not just at the end time.
+    def get_dt_step_controller(level):
+        step_ts = jnp.linspace(t0, t1, 2**level + 1, endpoint=True)
+        return None, diffrax.StepTo(ts=step_ts)
+
+    saveat = diffrax.SaveAt(ts=jnp.linspace(t0, t1, 2**level_coarse + 1, endpoint=True))
+
+    hs, errors, order = simple_sde_order(
+        bmkeys,
+        sde,
+        solver_ctr(),
+        ref_solver,
+        (level_coarse, level_fine),
+        get_dt_step_controller,
+        saveat,
+        bm_tol=2**-14,
+    )
+    # The upper bound needs to be 0.25, otherwise we fail.
+    # This still preserves a 0.05 buffer between the intervals
+    # corresponding to the different orders.
+    print(order)
+    assert -0.2 < order - theoretical_order < 0.25
+
+
+# Make variables to store the correct solutions in.
+# This is to avoid recomputing the correct solutions for every solver.
+solutions = {
+    "Ito": {
+        "any": None,
+        "com": None,
+        "add": None,
+    },
+    "Stratonovich": {
+        "any": None,
+        "com": None,
+        "add": None,
+    },
+}
+
+
+# Now compare the limit of Euler/Heun to the limit of the other solvers,
+# using a single reference solution. We use Euler if the solver is Ito
+# and Heun if the solver is Stratonovich.
+@pytest.mark.parametrize("solver_ctr,noise,theoretical_order", _solvers_and_orders())
+def test_sde_strong_limit(
```

Comment:
For these new tests, should we run them for all solvers or just the new SRK ones?

That is, how long do they take to run? Right now the Diffrax tests take about 18 minutes to run on a (my) laptop, and I'm cautious about increasing that.

Reply:
That's fair, I removed all the solvers that were there previously.